app/javascript/controllers/map_controller.js

```diff
@@ -7,7 +7,11 @@ export default class extends Controller {
   static targets = [ "canvas", "centreLatitude", "centreLongitude" ]
 
   initialize() {
+    console.log("Initializing MapController")
+
     this.map = new Map("map", chamonixValley)
+    this.vendorMarkers = []
+    const vendorMarkers = this.vendorMarkers
```

Comment:
You should be able to only use `this.vendorMarkers` or `vendorMarkers` as the second is just a reference to the first one. 
To use `this.vendorMarkers` the `setMarkers` function has to be an arrow function for right this context

Reply:
Aha, you're right. I'll try to change this. It might also depend on the controller lifecycle. (I'm still not getting that correct automatically)

Reply:
Aha, you're right. I'll try to change this. It might also depend on the controller lifecycle. (I'm still not getting that correct automatically)